src/WebJobs.Script.WebHost/Middleware/SystemTraceMiddleware.cs

```diff
@@ -34,7 +35,25 @@ public async Task Invoke(HttpContext context)
             string userAgent = context.Request.GetHeaderValueOrDefault("User-Agent");
             _logger.ExecutingHttpRequest(requestId, context.Request.Method, userAgent, context.Request.Path);
 
-            await _next.Invoke(context);
+            try
```

Comment:
This logic is overloading the `SystemTraceMiddleware` (its responsibility is to handle request tracing). Can you please move this to a new piece of middleware and have it added to the pipeline right after this one?

Reply:
That makes sense, I can do that!